test/e2e/specs/default-config/recordset/facet-within-facet.spec.ts

```diff
@@ -0,0 +1,740 @@
+import { test, expect, TestInfo, Page, Locator } from '@playwright/test';
+
+import ModalLocators from '@isrd-isi-edu/chaise/test/e2e/locators/modal';
+import RecordeditLocators from '@isrd-isi-edu/chaise/test/e2e/locators/recordedit';
+import RecordLocators from '@isrd-isi-edu/chaise/test/e2e/locators/record';
+import RecordsetLocators from '@isrd-isi-edu/chaise/test/e2e/locators/recordset';
+
+import { getCatalogID } from '@isrd-isi-edu/chaise/test/e2e/utils/catalog-utils';
+import {
+  testClearAllFilters, testDisplayedFacets, testSelectFacetOption,
+} from '@isrd-isi-edu/chaise/test/e2e/utils/recordset-utils';
+
+/**
+ * first layer facets:
+ *
+ * main
+ *     <-- main_inbound1
+ *                      <-- main_inbound1_inbound1 : "Entity path" and "Entity path with filter" facets
+ *                      <-- main_inbound1_inbound2: "Entity path 2" facet
+ *     <-- main_inbound2
+ *                      <-- main_inbound2_inbound1 : "Entity path with shared prefix"
+ *                      --> main_inbound2_outbound1: "Entity path with shared prefix and filter"
+ *     <-- main_inbound3
+ *                      <-- main_inbound3_inbound1: "Entity Path with fast filter"
+ *                      <-- main_inbound3_inbound1_fast_filter: the fast_fitler_source of prev facet
+ *
+ * To simplify the second layer facets, I added a facet_2nd_layer_hub which has fk to all the tables above,
+ * and other new tables have inbound to them to be displayed as facets. so regardless of which facet we opened
+ * for the first layer, the second layer facets all look the same in UI but they have different paths to the main table.
```

Comment:
These are the reasons for why I designed the tables this way:

1. I wanted to make sure that I can have two levels of facets, and on each level I wanted to have all the 6 different 
facets. 

2. All paths are using two hops because I wanted to force the `null` facet for all these paths to use the right outer join (we try to optimize the `null` facet and in some cases we might eliminate one hop so we can use inner join). All the bugs that I found were related to how the right outer join works, so I wanted to make sure that I test this for all the cases.

3. I wanted to reuse as many tables as possible so it's a bit easier to follow and test.


So the facets that are displayed for the first facet all are based on a 2-hop path to main. To better understand why I added `facet_2nd_layer_hub` let's focus on only two of the facets to show how this helped me reduce the size of schema. For example these two facets (using shorter name so it fits the page):

```
main <-- main_i1 <-- main_i1_i1

main <-- main_i2 <-- main_i2_i1
```

When you open the first facet, we're going to show the rows in `main_inbound1_inbound1`, and when you open the other one we will show the `main_inbound2_outbound1` table. If we want to design a facet for these tables similar to the first layer, we should create 2-hop paths from these tables to the facet that I want. For example I could do this:

```
main_i1_i1  <-- main_i1_i1_i1 <-- main_i1_i1_i1_i1

main_i1_i1  <-- main_i1_i1_2 <-- main_i1_i1_i2_i1

main_i2_i1 <-- main_i2_i1_i1 <-- main_i2_i1_i1_i1

main_i2_i1 <-- main_i2_i1_i2 <-- main_i2_i1_i2_o1
```

And this is just for 2 facets and showing 2 facets on the second layer. As you can imagine doing this for all 5 paths and having to create 5 other unique path for each one would be hard to manage. Since all I care about is having a 2-hop path, I added a `facet_2nd_layer_hub` in between, so the above could be simplified as just:

```
main_i1_i1 <-- facet_2nd_layer_hub
main_i2_i1 <-- facet_2nd_layer_hub

facet_2nd_layer_hub <-- facet_2nd_layer_hub_i1
facet_2nd_layer_hub <-- facet_2nd_layer_hub_i2
```

This way, the facets for the `main_i1_i1` are:

```
main_i1_i1 <-- facet_2nd_layer_hub <-- facet_2nd_layer_hub_i1
main_i1_i1 <-- facet_2nd_layer_hub <-- facet_2nd_layer_hub_i2
```

And the facets for `main_i2_i1`:

```
main_i2_i1 <-- facet_2nd_layer_hub <-- facet_2nd_layer_hub_i1
main_i2_i1 <-- facet_2nd_layer_hub <-- facet_2nd_layer_hub_i2
```